vss-extension.json

```diff
@@ -117,42 +117,76 @@
                     "displayName": "Marketplace URL",
                     "value": "https://marketplace.visualstudio.com"
                 },
-                "authenticationSchemes": [
+              "authenticationSchemes": [
+                {
+                  "type": "ms.vss-endpoint.endpoint-auth-scheme-token",
```

Comment:
So, I managed to check if it's allowed to save a service endpoint with an empty URL, and I confirm that a URL is always required. I confess that the approach of having two target URLs in the same endpoint is something that feels odd.  I've playing and trying different approaches, and I would like to suggest a different approach and hear your opinio - keep the current endpoint type as is, which is intended to target the Marketplace, the public (we can add a helpMarkDown to state that this for the public marketplace). Our tasks would stop asking for a specific type like we are doing `"type": "connectedService:VstsMarketplacePublishing"` and allow to specify any endpoint type `"type": "connectedService:"`. For On-Prem, users would just need to use a Generic endpoint with their basic auth (we can guide them in the input help). Then in the code `we could check if "username" was specified or not to decide which auth switch to use for tfx.